build.zig

```diff
@@ -321,7 +308,7 @@ pub fn build(b: *std.Build) !void {
         go_client(
             b,
             mode,
-            &.{ &install_step.step, &tb_client_header_generate.step },
+            &.{&install_step.step},
```

Comment:
I'm worried if it would impact the development cycle for the Go client, since now we need to regenerate the header in a separate step before building the Go library.

Couldn't we run the header test during `./zig/zig build scripts -- ci --language=go` as we do with the JNI test for the Java client?

Reply:
I don't think this would affect work on the go client much. tb_client.h is commited to the main branch, and, thanks to the test, every commit guarantees that it actually matches our C client. 

So, if you want to hack on go client in isolation, there's no need to manually trigger the rebuild, beacause the file is guaranteed to be fresh.

What might get harder are changes where you want to atomically change both the C client, and also the Go client. In this situation, you'd have to manually kick `./zig/zig build test -- "tb_client.h: generate"`. But this feels ok: it makes sense that, before hacking on clients, you want to make sure that the tests for TigerBeetle itself pass.

Reply:
> What might get harder are changes where you want to atomically change both the C client, and also the Go client.

Precisely, that's the most common workflow when implementing new features.

What I mean is that none of the other clients require an updated `tb_client.h` to work, only the binding code generated during CI is sufficient to make any change in the tb_client lib visible to them. Except for the GO client, in which the `tb_client.h` file is part of the binding code required by CGO.

If we don't want to _proactively_ generate the header during `zig build scripts -- ci—-language=go`, we should at least _preventively_ check if the header is up to date, warning the user to regenerate it, avoiding footguns. Still, I'm in favor of reducing the number of things one must remember to perform a task.

Reply:
Ah, I'd expect us to typically _not_ need atomic changes, so that "this is how we change C code, and that's the correspending header changes" and "here's how we adapt Go code to the new headers" are in separate commits, but it makes sense that sometimes its easier to do this in lock-step. 

I've added this test as a pre-requisite for go client build. This is a bit hacky, I think ideally we'd use the LazyPath infrastructure here, but I am not 100% sure where will this end up at the end of the larger build.zig refactor, so went with the simpler  solution for now!

Reply:
Ah, that doesn't trivially work, as our tests are incompatible with `-D=production`, need to think about how to do this in a nicer way...